### ISSUE TITLE
Sending additional data on Sign Up

### ISSUE THREAD

**Issue by jenweber** [ga-wdi-boston/capstone-project#100]
I have a drop-down menu that populates with an organization list now, but I'm not sure how to send the "name" and "organization_id" along with the email and password on signup. The username shows up when console.log the credentials, but not the organization_id. Neither of these two items make it into the database. What am I missing?

sign-up-form template

```
{{username-input name=credentials.name}}
{{email-input email=credentials.email}}
{{password-input password=credentials.password}}
{{password-confirmation-input password=credentials.passwordConfirmation}}
<div class="dropdown">
  <select name="organizations">
    {{#each organizations as |organization|}}
      <option value={{organization.id}}>{{organization.name}}</option>
    {{/each}}
  </select>
</div><br>
```

user model:

```
export default DS.Model.extend({
  email: DS.attr('string'),
  organization_id: DS.attr('integer'),
  name: DS.attr('string')
});
```

username-input template

```
<label for="name">Username</label>
{{input type="text"
        id="username"
        placeholder="The name your colleagues will see"
        value=name}}
```

**Comment by faetea** [comment]
Where in your database are you trying to send?
If `organization_id` is going into the users table then you will want to pass it along to the database at the same time as credentials.

So in `components/sign-up-form/component.js`, the submit action is only sending credentials, it isn't sending the dropdown selection.

```
actions: {
    submit () {
      this.sendAction('submit', this.get('credentials'));
    },
```

**Comment by jenweber** [comment]
I want to set the value=id from the dropdown menu as the organization_id for the user, and send that together with the rest of the credentials info like email and password. I understand that I'm not giving ember what it needs to know that organization_id = organization.id from the dropdown, but I'm not sure how to fix that.

Example working curl script on the back end:

```
curl --include --request POST http://localhost:3000/sign-up \
  --header "Content-Type: application/json" \
  --data '{
    "credentials": {
      "name":"Admin5",
      "email": "admin5@admin.com",
      "password": "admin",
      "password_confirmation": "admin",
      "organization_id":"2",
      "admin":true
    }
  }'
```

I got name to work, however, by adding it to the auth service.js:

```
  signUp (credentials) {
    console.log(credentials.id);
    return this.get('ajax').post('/sign-up', {
      data: {
        credentials: {
          email: credentials.email,
          name: credentials.name,
          password: credentials.password,
          password_confirmation: credentials.passwordConfirmation,
          organization_id: credentials.organization_id
        }
      },
    });
  },
```

**Comment by faetea** [comment]
I see you put `organization_id: credentials.organization_id`.  
What I mean is, in `components/sign-up-form/template.hbs`, all of the inputs have 
the `...name=credentials.name}}` or `...email=credentials.email}}` pattern 
and `components/sign-up-form/component.js` is using that `credentials: {},`.

If it was me, I would try generating a organization-dropdown component so that I could reference it as `{{organization-dropdown organization_id=credentials.organization_id}}`

Tho have you tried `value={{credentials.organization.id}}`, I'm not sure it will work, but that is something I would have tried as well.

**Comment by jenweber** [comment]
I will give these a shot and report back.

**Comment by jenweber** [comment]
I don't think i am able to pass the Organization data into the component at the same time as using the organization=credentials.organization format.

The code below does not populate the dropdown options:

```
<div class="dropdown">
  <select name="organizations">
    {{#each organizations as |organization|}}
      {{organization-input organization=credentials.organization }}
    {{/each}}
  </select>
</div><br>
```

Organization-input template:
`<option id="organization" value={{organization.id}}>{{organization.name}}</option>`

This code from closed Issue #97 may be relevant.

**Comment by GA-MEB** [comment]
@jenweber 
I'm a little confused. Your component is `sign-up-form`, correct? How do you reference it from its parent template? Data can only go into an Ember Component either (a) through parent template, or (b) by injecting a service. Which way is the list of organizations coming in?

Also, are these each their own component?

``` html
{{username-input name=credentials.name}}
{{email-input email=credentials.email}}
{{password-input password=credentials.password}}
{{password-confirmation-input password=credentials.passwordConfirmation}}
```

re: the dropdown, making dropdown menus work in Ember 2 isn't hard, but it is a little unintuitive (and less simple, unfortunately, than it was in Ember 1). 

Basically, you need to use Ember jQuery (`Ember.$`) to get value of the `<select>` element. 

To detect when the value has changed, add an action to the `<select>` that gets triggered on a `'change'` event. 

``` js
<select {{action 'selectValue' on='change'}}>
   // ...
</select>
```

**Comment by jrhorn424** [comment]
>  Basically, you need to use Ember jQuery (Ember.$) to value of the <select> element

I'd use the `ajax` service instead. It ships with `ember-cli`. The docs are at `ember-ajax`.

**Comment by jenweber** [comment]
Solved, how to send the data from a dropdown menu along with other Sign-Up form data, using Matt's method:

Add an action to the sign-up-form component.js:

```
    selectValue () {
      this.set('credentials.organization_id', Ember.$('select').val());
    },
```

Where organization_id is part of credentials on the back end. We're adding a key value pair to the user's credentials object, before the form is submitted.

In sign-up-form template:

```
  <select name="organizations" {{action "selectValue" on="change"}}>
    {{#each organizations as |organization|}}
      <option id="organization_id" value={{organization.id}}>{{organization.name}}</option>
    {{/each}}
  </select>
```

When the selected option value changes, the selectValue action is triggered.

This solution assumes that you have already solved the problem of getting the organization data into the form. See issue #97 if your dropdown/select menu is empty.
